src/avro/src/schema.rs

```diff
@@ -262,6 +271,23 @@ pub enum SchemaPiece {
     },
 }
 
+impl SchemaPiece {
+    /// Returns whether the schema node is "underlyingly" an Int (but possibly a logicalType typedef)
+    pub fn is_underlying_int(&self) -> bool {
```

Comment:
would `ResolveIntTs{Milli,Micro}` be an underlying int or an underlying long or neither?

Reply:
I guess "neither" or maybe "both" -- but this function is only called from schemas that have not been resolved yet, so it won't arise in practice. I know the API is not great... really what we probably want in the future is for Schema and ResolvedSchema to be different types so that you will never be calling methods (like this one) that only make sense for a schema on a resolved schema, but we're not there yet.